### PR TITLE
add eye op, kernel and unitest test=develop

### PR DESCRIPTION
**Pull request by ForFishes**
Add eye op implementation and the unit test codes of it. This op constructs an identity matrix or a batch of matrices.

### CODE REVIEW

File: paddle/fluid/operators/eye_op.cc [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc]: why not 2019?

File: paddle/fluid/operators/eye_op.cc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+namespace operators {
+
+class EyeOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    PADDLE_ENFORCE(ctx->HasOutput("Out"),
+                   "Output(Out) of EyeOP should not be null.");
+    auto num_rows = ctx->Attrs().Get<int64_t>("num_rows");
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc]: Is any reference to this attribute name `num_rows`? Is this common enough?

File: paddle/fluid/operators/eye_op.cc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    PADDLE_ENFORCE(num_rows >= 0,
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc:28]: why is it valid to be equal to `0`?

File: paddle/fluid/operators/eye_op.cc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                   "The value of Input(num_rows) should be non-negative int.");
+    auto num_columns = ctx->Attrs().Get<int64_t>("num_columns");
+    if (num_columns == -1) num_columns = num_rows;
+    PADDLE_ENFORCE(
+        num_columns >= 0,
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc:33]: the same problem with `num_rows`

File: paddle/fluid/operators/eye_op.cc [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    AddAttr<int64_t>("num_rows",
+                     "(int64_t) the number of rows in output matrix");
+    AddAttr<int64_t>("num_columns",
+                     "(int64_t) the number of columns in output matrix")
+        .SetDefault(-1);
+    AddOutput("Out",
+              "(Tensor) Construct an identity tensor with "
+              "specified shape");
+    AddComment(R"DOC(
+Return an identity tensor
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc]: lack of description about this op

File: paddle/fluid/operators/eye_op.cc [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+namespace ops = paddle::operators;
+using CPU = paddle::platform::CPUDeviceContext;
+using float16 = paddle::platform::float16;
+
+REGISTER_OPERATOR(eye, ops::EyeOp, ops::EyeOpMaker, ops::EyeOpVarTypeInference,
+                  paddle::framework::EmptyGradOpMaker);
+
+REGISTER_OP_CPU_KERNEL(eye, ops::EyeKernel<CPU, float>,
+                       ops::EyeKernel<CPU, double>,
+                       ops::EyeKernel<CPU, int64_t>, ops::EyeKernel<CPU, int>,
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cc:90]: inconsistent format

File: paddle/fluid/operators/eye_op.cu [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.cu]: why not 2019?

File: paddle/fluid/operators/eye_op.h [hunk 8]
```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (gavin1332) [inline paddle/fluid/operators/eye_op.h]: why not 2019?

File: python/paddle/fluid/tests/unittests/test_eye_op.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_eye_op.py]: why not 2019?

File: python/paddle/fluid/tests/unittests/test_eye_op.py [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    def setUp(self):
+        '''
+	Test eye op with specified shape
+        '''
+        self.op_type = "eye"
+
+        self.inputs = {}
+        self.attrs = {
+            'num_rows': 2019,
+            'num_columns': 3019,
```
Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_eye_op.py]: is the number a little large for an ut?

File: python/paddle/fluid/tests/unittests/test_eye_op.py [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        '''
+        self.op_type = "eye"
+
+        self.inputs = {}
+        self.attrs = {'num_rows': 2019, 'num_columns': 1}
+        self.outputs = {'Out': np.eye(2019, 1, dtype=float)}
+
+    def test_check_output(self):
+        self.check_output()
+
```
Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_eye_op.py:71]: missing test for `batch_shape`

**Review by gavin1332 — CHANGES_REQUESTED** [review]
fix copyright and response to the questions

**Comment by ForFishes** [comment]
MacHi 2019-08-01 21-09-10

**Comment by ForFishes** [comment]
MacHi 2019-08-01 21-09-10

**Pull request closed without merging**